File: Java-Kotlin/src/main/java/jianzhi/jz1.java
package jianzhi;

public class jz1 {

    private static boolean isValid(int[][] array) {
        return array != null && array.length != 0 && array[0].length != 0;
    }

    public boolean Find(int target, int[][] array) {
        if (!isValid(array)) {
            return false;
        }

        int rows = array.length;
        int cols = array[0].length;
        int row = rows - 1;
        int col = 0;
        while (row >= 0 && col < cols) {
            if (array[row][col] < target) {
                col++;
            } else if (array[row][col] > target) {
                row--;
            } else {
                return true;
            }
        }
        return false;
    }
}

File: Go-Solutions/nowcoder/jz1.go
package nowcoder

func Find(target int, array [][]int) bool {
	// write code here
	if !isValidjz1(array) {
		return false
	}

	rows, cols := len(array), len(array[0])
	row, col := rows-1, 0
	for row >= 0 && col < cols {
		if array[row][col] < target {
			col++
		} else if array[row][col] > target {
			row--
		} else {
			return true
		}
	}

	return false
}

func isValidjz1(array [][]int) bool {
	if array == nil || len(array) == 0 || len(array[0]) == 0 {
		return false
	}
	return true
}
